File: backend-ms1/src/main/java/ma/zs/stocky/service/facade/admin/produit/CategorieProduitAdminService.java
package ma.zs.stocky.service.facade.admin.produit;

import java.util.List;
import ma.zs.stocky.bean.core.produit.CategorieProduit;
import ma.zs.stocky.dao.criteria.core.produit.CategorieProduitCriteria;
import ma.zs.stocky.zynerator.service.IService;



public interface CategorieProduitAdminService extends  IService<CategorieProduit,CategorieProduitCriteria>  {

    List<CategorieProduit> findByCompteBudgetaireId(Long id);
    int deleteByCompteBudgetaireId(Long id);
    long countByCompteBudgetaireId(Long id);




}

File: frontend/src/app/shared/service/admin/produit/CategorieProduitAdmin.service.ts
import { Injectable } from '@angular/core';
import {HttpClient} from '@angular/common/http';
import {BehaviorSubject, Observable} from 'rxjs';

import {environment} from 'src/environments/environment';

import {CategorieProduitDto} from 'src/app/shared/model/produit/CategorieProduit.model';
import {CategorieProduitCriteria} from 'src/app/shared/criteria/produit/CategorieProduitCriteria.model';
import {AbstractService} from 'src/app/zynerator/service/AbstractService';


@Injectable({
  providedIn: 'root'
})
export class CategorieProduitAdminService extends AbstractService<CategorieProduitDto, CategorieProduitCriteria> {
     constructor(private http: HttpClient) {
        super();
        this.setHttp(http);
    }

    public constrcutDto(): CategorieProduitDto {
        return new CategorieProduitDto();
    }

    public constrcutCriteria(): CategorieProduitCriteria {
        return new CategorieProduitCriteria();
    }

    get API() {
        return environment.apiUrlStockyservice + 'admin/categorieProduit/';
    }
}
